akka-stream-tests/src/test/scala/akka/stream/scaladsl/FromMaterializationSpec.scala

```diff
@@ -61,7 +62,7 @@ class FromMaterializerSpec extends StreamSpec {
         }
         .named("my-name")
 
-      source.runWith(Sink.head).futureValue shouldBe Some("setup-setup-my-name")
+      source.runWith(Sink.head).futureValue shouldBe Some("setup-my-name-setup")
```

Comment:
@johanandren You asked about this in #30142. In fact, the correct order is `setup-my-name-setup`as the names are concatenated in the order of least-specific to most-specific:
1. The outer `fromMaterializer` is named `setup`
2. The outer `fromMaterializer` is then named `my-name`, which has higher specificity
3. The inner `fromMaterializer` is named `setup`